cmd/oras/internal/option/remote.go

```diff
@@ -243,6 +250,21 @@ func (opts *Remote) authClient(registry string, debug bool) (client *auth.Client
 	return
 }
 
+func (opts *Remote) parseCustomHeaders() error {
+	if len(opts.headerFlags) != 0 {
+		headers := map[string][]string{}
+		for _, h := range opts.headerFlags {
+			name, value, found := strings.Cut(h, ":")
+			if !found || strings.TrimSpace(value) == "" {
```

Comment:
According to [RFC 2616 Section 4.2](https://www.rfc-editor.org/rfc/rfc2616#section-4.2), the name must not be empty and the value can be empty.

Reply:
Note:
```go
package main

import (
	"net/http"
	"os"
)

func main() {
	headers := make(http.Header)
	headers[""] = []string{"abc"}
	headers["    "] = []string{"efg"}
	headers["empty"] = []string{""}
	headers["space"] = []string{"   "}
	headers["foo"] = []string{"bar"}
	headers["foo2"] = []string{"  bar2"}

	headers.Write(os.Stdout)
}
```
outputs
```
empty: 
foo: bar
foo2: bar2
space: 
```

Reply:
> According to [RFC 2616 Section 4.2](https://www.rfc-editor.org/rfc/rfc2616#section-4.2), the name must not be empty and the value can be empty.

curl's behavior is the other way around. Which should I follow, curl or the RFC spec?

Reply:
Always follow the spec since the spec is the standard.

Reply:
Changed the behavior of parsing to be conformant to the spec.